### PR TITLE
ramips: add support for TP-Link TL-WR850N v2

### PR DESCRIPTION
**Pull request by inflabz**
This patch adds support for the TP-Link TL-WR850N v2, ported from the ramips target.
This router is very similar to TP-Link TL-WR840 v4 and TP-Link TL-WR841 v13

Specifications:
SOC: MediaTek MT7628NN
Flash: 8 MiB SPI
RAM: 64 MiB
WLAN: MediaTek MT7628NN
Ethernet: 5 ports (100M)

Installation:
Using the integrated tftp capability of the router.
Turn off the router.
Connect pc to one of the router LAN ports.
Set your PC IPv4 address to 192.168.0.66/24.
Run any TFTP server on the PC.
Put the recovery firmware on the root directory of TFTP server and name the file tp_recovery.bin (renamed from tftp-recovery.bin)
Start the router by pressing power button while holding the WPS/Reset button (or both WPS/Reset and WIFI buttons)
Router connects to your PC with IPv4 address 192.168.0.2, downloads the firmware, installs it and reboots. LEDs are flashing. Now you have OpenWrt installed.
Change your IPv4 PC address to something in 192.168.1.0/24 network or use DHCP to get an address from your OpenWrt router.
Done! You can login to your router via ssh.

Forum link https://forum.openwrt.org/t/add-support-for-tp-link-tl-wr850n-v2/66899

### CODE REVIEW

**Comment by adrianschmutzler** [comment]
> Installation is identical to tp link tl-wr840 v4 tftp recovery installation instructions

Then please provide these in the commit message.

File: target/linux/ramips/dts/mt7628an_tplink_tl-wr850n-v2.dts [hunk 1]
```diff
@@ -0,0 +1,65 @@
+/dts-v1/;
```
Comment (adrianschmutzler) [inline target/linux/ramips/dts/mt7628an_tplink_tl-wr850n-v2.dts:1]: Please add an SPDX license identifier.

File: target/linux/ramips/image/mt76x8.mk [hunk 2]
```diff
@@ -474,6 +474,20 @@ define Device/tplink_tl-wr842n-v5
...
+  $(Device/tplink-v2)
+  IMAGE_SIZE := 7808k
+  DEVICE_MODEL := TL-WR850N
+  DEVICE_VARIANT := v2
+  TPLINK_FLASHLAYOUT := 8Mmtk
+  TPLINK_HWID := 0x08500002
+  TPLINK_HWREVADD := 0x2
+  IMAGES := sysupgrade.bin tftp-recovery.bin
+  IMAGE/tftp-recovery.bin := pad-extra 128k | $$(IMAGE/factory.bin)
+  SUPPORTED_DEVICES += tl-wr850n-v2
```
Comment (adrianschmutzler) [inline target/linux/ramips/image/mt76x8.mk:487]: This line should be dropped.

File: target/linux/ramips/mt76x8/base-files/etc/board.d/01_leds [hunk 3]
```diff
@@ -81,7 +81,8 @@ tplink,tl-wr842n-v5)
 	ucidef_set_led_switch "lan" "lan" "$boardname:green:lan" "switch0" "0x1e"
 	ucidef_set_led_switch "wan" "wan" "$boardname:green:wan" "switch0" "0x01"
 	;;
-tplink,tl-wr840n-v4)
+tplink,tl-wr840n-v4|\
+tplink,tl-wr850n-v2)
 	ucidef_set_led_wlan "wlan2g" "wlan2g" "$boardname:green:wlan" "phy0tpt"
```
Comment (adrianschmutzler) [inline target/linux/ramips/mt76x8/base-files/etc/board.d/01_leds:86]: Please use a DT-trigger instead of this line:
`linux,default-trigger = "phy0tpt";`

**Comment by adrianschmutzler** [comment]
The dot at the end of the commit title should be removed.

Have you checked whether the MAC assignment matches the one on vendor OS?

Does the device have a label with a MAC address on it? To which interface does it correspond?

File: target/linux/ramips/image/mt76x8.mk (hunk 2 above)
Comment (inflabz) [reply target/linux/ramips/image/mt76x8.mk:487]: I did not understand which line should be dropped?

File: target/linux/ramips/dts/mt7628an_tplink_tl-wr850n-v2.dts (hunk 1 above)
Comment (inflabz) [reply target/linux/ramips/dts/mt7628an_tplink_tl-wr850n-v2.dts:2]: added

**Comment by inflabz** [comment]
> The dot at the end of the commit title should be removed.
> 
> Have you checked whether the MAC assignment matches the one on vendor OS?
> 
> Does the device have a label with a MAC address on it? To which interface does it correspond?

The MAC address on the router case and on the eth0 interface matches

File: target/linux/ramips/mt76x8/base-files/etc/board.d/01_leds (hunk 3 above)
Comment (inflabz) [reply target/linux/ramips/mt76x8/base-files/etc/board.d/01_leds]: made changes

File: target/linux/ramips/image/mt76x8.mk (hunk 2 above)
Comment (adrianschmutzler) [reply target/linux/ramips/image/mt76x8.mk:487]: SUPPORTED_DEVICES += tl-wr850n-v2

File: target/linux/ramips/mt76x8/base-files/etc/board.d/01_leds (hunk 3 above)
Comment (adrianschmutzler) [reply target/linux/ramips/mt76x8/base-files/etc/board.d/01_leds]: No, please.

First, this also changes the config for 840n-v4, which should stay unaffected.

What I asked you is to implement the WLAN LED by adding the trigger in the DTS file. You can grep for the line I've provided in the dts directory, and have a look how this is done for other devices. The ucidef_set_led_wlan line will then just be dropped for _your_ device.

**Comment by adrianschmutzler** [comment]
> The MAC address on the router case and on the eth0 interface matches

Just for documentation: The label MAC address is already set in the DTSI, so we don't need to do anything here.

File: target/linux/ramips/mt76x8/base-files/etc/board.d/01_leds (hunk 3 above)
Comment (inflabz) [reply target/linux/ramips/mt76x8/base-files/etc/board.d/01_leds]: Now have I done everything right?

Comment (adrianschmutzler) [reply target/linux/ramips/mt76x8/base-files/etc/board.d/01_leds]: You've done the right regarding the WiFi LED, but now you miss setup for lan/wan LEDs.

Comment (inflabz) [reply target/linux/ramips/mt76x8/base-files/etc/board.d/01_leds]: You're right
I fixed it

**Comment by inflabz** [comment]
OpenWrt Bootlog https://pastebin.com/6ahN0w6V

**Comment by adrianschmutzler** [comment]
You were already close, it just didn't work because you created another merge commit for the inflabz branch.

**Comment by darshkpatel** [comment]
Hey! Is there any way i can help this router have an official firmware?

**Comment by adschm** [comment]
As it appears, the initial author mostly had problems with handling git. Nothing has happened for two months here, so if you are interested, just pick up what's there, bring it in proper shape (squash into one commit, add proper description) and submit a new PR for it. No developer with commit access will care about a closed PR that does not meet formal requirements.

**Comment by darshkpatel** [comment]
> As it appears, the initial author mostly had problems with handling git. Nothing has happened for two months here, so if you are interested, just pick up what's there, bring it in proper shape (squash into one commit, add proper description) and submit a new PR for it. No developer with commit access will care about a closed PR that does not meet formal requirements.

I've opened a new PR after squashing all the commits and merging with the latest master

**Pull request closed without merging**